ledger/ledger-api-auth/src/test/suite/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserverSpec.scala

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.auth
+
+import java.time.{Clock, Duration, Instant, ZoneId}
+
+import akka.actor.{Cancellable, Scheduler}
+import com.daml.clock.AdjustableClock
+import com.daml.error.ErrorsAssertions
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.ledger.participant.state.index.v2.UserManagementStore
+import com.daml.logging.LoggingContext
+import com.daml.platform.server.api.validation.ErrorFactories
+import io.grpc.StatusRuntimeException
+import io.grpc.stub.ServerCallStreamObserver
+import org.mockito.{ArgumentCaptor, ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+class OngoingAuthorizationObserverSpec
+    extends AsyncFlatSpec
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar
+    with ErrorsAssertions {
+
+  private val loggingContext = LoggingContext.ForTesting
+
+  it should "signal onError aborting the stream when user rights state hasn't been refreshed in a timely manner" in {
```

Comment:
Shouldn't we have a test case for claims changing over time?

Reply:
That's tested in `OngoingStreamAuthIT`

Reply:
That's correct but the comments of the `OngoingAuthorizationObserver` class state that the class will ensure stream termination on changing user rights state. Even if that would be a test duplication, I'd still add such test case to `OngoingAuthorizationObserverSpec`.